server/app/assets/javascripts/admin_export_view.ts

```diff
@@ -1,20 +1,49 @@
 import {assertNotNull} from './util'
 
 class AdminExportView {
-  private static JSON_SELECTOR = 'program-json'
-  private static COPY_BUTTON_SELECTOR = 'copy-json-button'
+  // These values should be kept in sync with views/admin/migration/AdminExportView.java.
+  private static PROGRAM_EXPORT_FORM_ID = 'program-export-form'
+  private static GENERATE_JSON_BUTTON_ID = 'generate-json-button'
+  // These values should be kept in sync with views/admin/migration/AdminExportViewPartial.java.
+  private static PROGRAM_JSON_ID = 'program-json'
+  private static COPY_BUTTON_ID = 'copy-json-button'
 
   constructor() {
-    document.addEventListener('htmx:afterSwap', function () {
+    this.addSelectListenerToForm()
+    this.addClickListenerToCopyButton()
+  }
+
+  addSelectListenerToForm() {
+    const programExportForm = document.getElementById(
+      AdminExportView.PROGRAM_EXPORT_FORM_ID,
+    )
+    if (programExportForm) {
+      const generateJsonButton = assertNotNull(
+        document.getElementById(AdminExportView.GENERATE_JSON_BUTTON_ID),
+      ) as HTMLButtonElement
+      programExportForm.addEventListener('change', () => {
+        const options = document.querySelectorAll('input[name="programId"]')
+        // The "Generate JSON" button is disabled by default
+        // If any of the options are selected we want to enable the button
+        options.forEach((option) => {
+          const radioButton = option as HTMLInputElement
+          if (radioButton.checked) {
+            generateJsonButton.disabled = false
+          }
+        })
+      })
+    }
+  }
+
+  addClickListenerToCopyButton() {
+    document.addEventListener('htmx:afterSwap', () => {
       // htmx:afterSwap fires after htmx swaps in new content
       // the json and copy button are only availabe once htmx has swapped them in
```

Comment:
(suggestion) Make this a jsdoc comment for the method

Reply:
done